### ISSUE TITLE
 Best Compression  algorithm of  zlib  cannot be used

### ISSUE THREAD

**Issue by colorworlds** [godotengine/godot#40233]
**Godot version:**
3.2.2

**OS/device including version:**
win7

**Issue description:**
Setting the compression level  in the Project Settings is invalid for  `compress()/decompress()` method of PoolByteArray 

**Steps to reproduce:**
First of all, this is the default for the zlib compression algorithm:
```
func _ready() -> void:
	var data:PoolByteArray = '{"body":[],"header_len":16,"op":5,"packet_len":634}'.to_utf8()      # Arbitrary data for compression
	var compress_data = data.compress(File.COMPRESSION_DEFLATE)        # compressed data
	print('%x|%x' % [compress_data[0],compress_data[1] ] )         # Print out ZLIB headers
```
```
Print Result:
78|9c
```
According to this link:
https://stackoverflow.com/questions/9050260/what-does-a-zlib-header-look-like
78|9c is the magic headers for Default Compression of zlib.

But I want to use the Best Compression .So I find "Project Settings->Compression->Fomats->Zlib" and set the "Compression Level" to 7.
 I set it according to this table:
```
ZLIB/GZIP headers
Level | ZLIB  | GZIP 
  1   | 78 01 | 1F 8B 
  2   | 78 5E | 1F 8B 
  3   | 78 5E | 1F 8B 
  4   | 78 5E | 1F 8B 
  5   | 78 5E | 1F 8B 
  6   | 78 9C | 1F 8B 
  7   | 78 DA | 1F 8B 
  8   | 78 DA | 1F 8B 
  9   | 78 DA | 1F 8B 
```
then repeat the code above:
```
func _ready() -> void:
	var data:PoolByteArray = '{"body":[],"header_len":16,"op":5,"packet_len":634}'.to_utf8()
	var compress_data = data.compress(File.COMPRESSION_DEFLATE)
	print('%x|%x' % [compress_data[0],compress_data[1] ] )
```
```
Print Result:
78|9c
```
I think this result should be 78 DA, not 78 9C. So the Project Settings do not work on `compress() `method
Is there a mistake in my understanding? or is it a bug? I'm not sure.

**Minimal reproduction project:**
The code has been posted on it. It's really short.

**Comment by mrushyendra** [comment]
It appears that the compression level is set only once initially in the following code:
https://github.com/godotengine/godot/blob/f10f76d95db3248ffb11d2b97b0cdd317b0b5e73/core/project_settings.cpp#L1209-L1218

Updating the compression level using the Project Settings Menu, or via script thus has no effect on the compression level, since `compress()` does not use  `GLOBAL_GET` to get the latest compression level.

**Comment by colorworlds** [comment]
@mrushyendra  So, In terms of design, should the `compress()` function be affected by the compression level ? I don't know much about the role of compression levels in project settings.

**Comment by Xrayez** [comment]
Yeah, I've stumbled upon this some years ago but wasn't sure whether that's a bug, and now I've figured out that this is a regression which dates back to 2017: bbada82f8088e18b663f087484fcbb662dd76a5a.

Should be fixed by #40340.
